### PR TITLE
Update pyproject.toml for Poetry mismatch

### PR DESCRIPTION
**Pull request by agamm**
## Why are these changes needed?

Poetry install fails without these changes.

## Related issue number

Fixes https://github.com/microsoft/autogen/issues/1426

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by gagb** [comment]
I thought v0.2.7 claimed support for python 3.12:

https://github.com/microsoft/autogen/releases/tag/v0.2.7

(cc. @ekzhu )

**Comment by victordibia** [comment]
Good point @gagb .
@agamm , what version of pyautogen did you have installed ? Also, can you confirm if the poetry install error occurs in a new conda env? There is a chance you might have had an older version of autogen installed, but the recent version does not have the resolution error above.

File: samples/apps/autogen-studio/pyproject.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ authors = [
 description = "AutoGen Studio"
 readme = "README.md"
 license = { file="LICENSE" }
-requires-python = ">=3.9"
+requires-python = ">=3.9, <3.12"
```
Comment (agamm) [inline samples/apps/autogen-studio/pyproject.toml:13]:
```suggestion
requires-python = ">=3.9, <3.13"
```

**Review by agamm — COMMENTED** [review]
Would this be a solution?

**Review by victordibia — APPROVED** [review]

**Review by ekzhu — APPROVED** [review]

**Pull request merged**